### ISSUE TITLE
ErrorShape is not applied

### ISSUE THREAD

**Issue by johnny-y-wang** [KevinEdry/nestjs-trpc#36]
It seems like `errorShape` [as defined in this line](https://github.com/KevinEdry/nestjs-trpc/blob/5073129596ed27eafc08ccdfed242631cb06013b/packages/nestjs-trpc/lib/trpc.driver.ts#L62) is a no-op?

Per [tRPC docs](https://trpc.io/docs/server/error-formatting), the error formatter argument should be called `errorFormatter`, not `errorShape`. I might be missing something obvious here, but the disabled type-checking on this method does not inspire confidence :)

**Comment by KevinEdry** [comment]
Thats a good catch, its a trivial fix if you want to open a PR for it, if not I don't mind to do it instead.
I disabled typescript on it because I am implementing a dynamic router with no actual types (since the router is built in run-time from your nestjs providers) which causes a typescript mismatch here. I might want to add a typeguard instead here to have some type-safety.

**Comment by hrmcdonald** [comment]
I noticed this as well and planned on opening a ticket. I have tested this locally and `errorShape` is just the wrong property. After renaming it to `errorFormatter` in the code it works as intended.

---

This might be worth a separate issue, but I'd actually much prefer [another option altogether to enable implementing an `onError` handler](https://trpc.io/docs/server/error-handling#handling-errors) that allows us to provide a service so that we can hook into dependency injection to log errors for example. Essentially exactly how `context` works today with `TRPCContext` services. The same setup asking us to provide a `TRPCErrorHandler` would be great.

This would just need to be provide like the context in the drivers:
- [Express](https://github.com/KevinEdry/nestjs-trpc/blob/5073129596ed27eafc08ccdfed242631cb06013b/packages/nestjs-trpc/lib/drivers/express.driver.ts#L19)
- [Fastify](https://github.com/KevinEdry/nestjs-trpc/blob/5073129596ed27eafc08ccdfed242631cb06013b/packages/nestjs-trpc/lib/drivers/fastify.driver.ts#L19)

**Comment by KevinEdry** [comment]
> I noticed this as well and planned on opening a ticket. I have tested this locally and `errorShape` is just the wrong property. After renaming it to `errorFormatter` in the code it works as intended.
> 
> This might be worth a separate issue, but I'd actually much prefer [another option altogether to enable implementing an `onError` handler](https://trpc.io/docs/server/error-handling#handling-errors) that allows us to provide a service so that we can hook into dependency injection to log errors for example. Essentially exactly how `context` works today with `TRPCContext` services. The same setup asking us to provide something like a `TRPCErrorHandler` would be great.
> 
> This would just need to be provided like the context in the drivers:
> 
> * [Express](https://github.com/KevinEdry/nestjs-trpc/blob/5073129596ed27eafc08ccdfed242631cb06013b/packages/nestjs-trpc/lib/drivers/express.driver.ts#L19)
> * [Fastify](https://github.com/KevinEdry/nestjs-trpc/blob/5073129596ed27eafc08ccdfed242631cb06013b/packages/nestjs-trpc/lib/drivers/fastify.driver.ts#L19)

Thats defiantly doable, feel free to create a new issue for it so I can label it correctly, you are more than welcome to implement it yourself (I could always use some help 😅).

**Comment by KevinEdry** [comment]
> @KevinEdry Quick [PR](https://github.com/KevinEdry/nestjs-trpc/pull/37) on your way to fix this. Once published, could you also bump npm version so I can update my local repo?
> 
> @hrmcdonald I agree. That'd be nice!
> 
> And, in case I haven't mentioned, big thank you to @KevinEdry for this awesome repo! Wanted to use tRPC and NestJS together for years now and this is finally possible :D

Yup, once it is merged i'll bump the version.
Thanks for the kind words ❤️

**Comment by KevinEdry** [comment]
Released in
🔗 https://github.com/KevinEdry/nestjs-trpc/releases/tag/1.6.1
🔗 https://www.npmjs.com/package/nestjs-trpc
